jetty-http/src/main/java/org/eclipse/jetty/http/HttpURI.java

```diff
@@ -108,16 +134,25 @@ public HttpURI(String scheme, String host, int port, String path, String param,
         _scheme = scheme;
         _host = host;
         _port = port;
-        _path = path;
+        parse(State.PATH, path, 0, path.length());
```

Comment:
`path` may be null, so we need a null guard or we NPE -- I think passing `null` as path should be accepted.